lib/Echidna/Campaign.hs

```diff
@@ -187,21 +188,29 @@ callseq v w ql = do
   -- our execution function appropriately
   coverageEnabled <- isJust . knownCoverage <$> view hasLens
   let ef = if coverageEnabled then execTxOptC else execTx
+      old = v ^. env . EVM.contracts
   -- Then, we get the current campaign state
   ca <- use hasLens
   -- Then, we generate the actual transaction in the sequence
   is <- replicateM ql (evalStateT genTxM (w, ca ^. genDict))
   -- We then run each call sequentially. This gives us the result of each call, plus a new state
   (res, s) <- runStateT (evalSeq v ef is) (v, ca)
+  let new = s ^. _1 . env . EVM.contracts
+      diff = keys $ new \\ old
+      diffs = H.fromList [(AbiAddressType, S.fromList $ AbiAddress . addressWord160 <$> diff)]
   -- Save the global campaign state (also vm state, but that gets reset before it's used)
   hasLens .= snd s
   -- Now we try to parse the return values as solidity constants, and add then to the 'GenDict'
-  modifying (hasLens . constants) . H.unionWith (++) . parse res =<< use (hasLens . rTypes) where
+  types <- use $ hasLens . rTypes
```

Comment:
Can you add a few comments in `callseq` to explain how the address collection works?